app/code/Magento/Catalog/Block/Product/View.php

```diff
@@ -88,6 +94,7 @@ public function __construct(
         \Magento\Customer\Model\Session $customerSession,
         ProductRepositoryInterface $productRepository,
         \Magento\Framework\Pricing\PriceCurrencyInterface $priceCurrency,
+        \Magento\Catalog\Model\ProductFactory $productFactory,
```

Comment:
new dependency should be added last one to the class , even after $data array

and should be optional, default value is null

this is all done to prevent backward incompatibility